_pages/exercises/AutonomousCars/end_to_end_visual_control.md

```diff
@@ -0,0 +1,414 @@
+standard range, which improves training stability. The <strong>labels</strong> are extracted directly from the CSV files and linked to the appropriate images. To enhance <strong>robustness</strong> and <strong>generalization</strong>, <strong>data augmentation techniques</strong> 
+such as <strong>flipping</strong>, <strong>rotation</strong>, and <strong>brightness adjustment</strong> are applied, 
+creating diverse variations of the training data.
+</p>
+
+#### ii) Dataset Splitting
+
+<p style="text-align:justify">
+The <strong>Simple Circuit dataset</strong> is divided into three sets: <code class="language-plaintext highlighter-rouge" style="color:#222831;background:#bdbdbd">training</code> and <code class="language-plaintext highlighter-rouge" style="color:#222831;background:#bdbdbd">testing</code>. The training set contains the majority of the data and is used to teach the model to recognize patterns and learn the task. The testing set is the data used during training to fine-tune hyper-parameters, prevent over-fitting, and monitor the model’s performance on unseen data.
+</p>
+
+<p style="text-align:justify;">
+The <strong>Combine Circuit Dataset</strong> consists of <code class="language-plaintext highlighter-rouge" style="color:#222831;background:#bdbdbd">training</code> data and <code class="language-plaintext highlighter-rouge" style="color:#222831;background:#bdbdbd">adjustment data</code> (used for sharp turns). If needed, these can be combined to create a larger dataset. You can then split the combined dataset into <strong>training</strong> and <strong>testing</strong> sets using an <code class="language-plaintext highlighter-rouge" style="color:#222831;background:#bdbdbd">80%-20% ratio</code> with the <code class="language-plaintext highlighter-rouge" style="color:#222831;background:#bdbdbd">train_test_split</code> function <code class="language-plaintext highlighter-rouge" style="color:#222831;background:#bdbdbd">(from sklearn.model_selection)</code>.
+</p>
+<!-- <br> -->
+<p style="margin-bottom:4px; font-size:16px">Example:</p>
+
+```python
+from sklearn.model_selection import train_test_split
+
+# Combine training and adjustment data
+all_data = train_data + adjustment_data
+
+# Split into 80% train, 20% validation
+train_set, val_set = train_test_split(all_data, test_size=0.2, random_state=42)
+```
+
+#### iii) Model Architecture and Training:
+
+<p style="text-align:justify">
+For this exercise, a deep learning model, such as a Convolutional Neural Network (CNN) like <strong>PilotNet</strong> or <strong>ResNet</strong>, is chosen to perform End-to-End Visual Control. The model consists of multiple convolutional layers that automatically extract meaningful features from raw camera images, followed by fully connected layers that map these features to the predicted outputs, such as linear and angular velocities for the autonomous vehicle.
```

Comment:
I would point here to torchvision pretrained models as a starting point. Also link to PilotNet.